all_notams.py

```diff
@@ -2,41 +2,136 @@
+    
+    # Set the longitude and latitude to the beginning coordinates 
+    lon = coord1[1]
+    lat = coord1[0]
+
+    # Print out the starting and ending to ensure no infinite loop
+    print(f"Starting lat: {lat} and lon: {lon}")
+    print(f"Ending lat: {coord2[0]} and lon: {coord2[1]}")
+
+    for index in range(int(mile_step)):
+        # Print the current coordinates
+        print(f"Now on latitude: {lat} and longitude: {lon}")
+
+        # Set the url to the FAA API
+        url = f'{API_URL}locationLongitude={lon}&locationLatitude={lat}'
+        headers = {'client_id' : secret.client_id, 'client_secret' : secret.client_secret}
+        raw_json = requests.get(url, headers=headers).json()
+
+        # Get the total number of pages in case we must deal with pagination
+        total_pages = raw_json["totalPages"]
+
+        # For loop to iterate through each page
+        for index in range(1, total_pages + 1):
+            paged_url = f"{API_URL}locationLongitude={lon}&locationLatitude={lat}&pageNum={index}"
+            json_string = requests.get(paged_url, headers=headers).json()
+            items = json_string["items"]
+            
+            # Iterate through each NOTAM returned from the API call
+            for entry in items:
+
+                # If the first NOTAM is not international then it is in local format
+                if(entry["properties"]["coreNOTAMData"]["notamTranslation"][0]["type"] != "ICAO"):
+                    entryKey = entry["properties"]["coreNOTAMData"]["notam"]["number"]
+                    dictionary[entryKey] = entry
+
+        # Increment by the step
+        lon += lon_step
+        lat += lat_step
+        time.sleep(2)
+
+    # Set the longitude and latitude to the final coordinates
+    lon = coord2[1]
+    lat = coord2[0]
+
+    # Set the url to the FAA API
+    url = f'{API_URL}locationLongitude={lon}&locationLatitude={lat}'
+    headers = {'client_id' : secret.client_id, 'client_secret' : secret.client_secret}
+    raw_json = requests.get(url, headers=headers).json()
+
+     # Get the total number of pages in case we must deal with pagination
+    total_pages = raw_json["totalPages"]
+
+    # For loop to iterate through each page
+    for index in range(1, total_pages + 1):
+        paged_url = f"{API_URL}locationLongitude={lon}&locationLatitude={lat}&pageNum={index}"
+        json_string = requests.get(paged_url, headers=headers).json()
+        items = json_string["items"]
+            
+        # Iterate through each NOTAM returned from the API call
+        for entry in items:
+
+            # If the first NOTAM is not international then it is in local format
+            if(entry["properties"]["coreNOTAMData"]["notamTranslation"][0]["type"] != "ICAO"):
+                entryKey = entry["properties"]["coreNOTAMData"]["notam"]["number"]
+                dictionary[entryKey] = entry
+
+    return dictionary
+
+
+def main():
```

Comment:
```suggestion
if __name__ == "__main__":
```

You can look this up, but this is how python does main functions. If you do this and delete the call to `main()` at the end of the file, this main function will still run when writing `python all_notams.py`, but when it is imported as a module to another script, main will not run. The reason I was getting double calls to the API was because your main function was written like this and was being called everytime this script was imported into my program.